scrc/preprocessors/extractors/spider_specific/court_composition_extracting_functions.py

```diff
@@ -200,3 +200,423 @@ def prepare_french_name_and_find_gender(person: CourtPerson) -> CourtPerson:
+    if start_pos:
+        header = header[start_pos.span()[0]:]
+
+    information_end_regex = r'Urteil|U R T E I L'
+    end_pos = re.search(information_end_regex, header)
+    if end_pos:
+        header = header[:end_pos.span()[1] - 1]
+
+    header = header.replace(';', ',')
+    header = header.replace(' und ', ', ')
+    header = header.replace(' sowie ', ', ')
+    header = header.replace('lic. ', '')
+    header = header.replace('iur. ', '')
+    header = header.replace('Dr. ', '')
+    header = header.replace('MLaw ', '')
+    header = header.replace('PD ', '')
+    header = header.replace(' als Einzelrichterin', '')
+    header = header.replace(' als Einzelrichter', '')
+    besetzungs_strings = header.split(',')
+
+
+    besetzung = CourtComposition()
+    current_role = CourtRole.JUDGE
+    last_person: CourtPerson = None
+    person: CourtPerson = None
+    last_gender = Gender.MALE
+
+    personal_information_database = json.loads(Path("personal_information.json").read_text())
+
+    def match_person_to_database(person: CourtPerson, current_gender: Gender) -> Tuple[CourtPerson, bool]:
+        """"Matches a name of a given role to a person from personal_information.json"""
+        results = []
+        name = person.name.replace('.', '').strip()
+        split_name = name.split()
+        initial = False
+        if len(split_name) > 1:
+            initial = next((x for x in split_name if len(x) == 1), None)
+            split_name = list(filter(lambda x: len(x) > 1, split_name))
+        if person.court_role.value in personal_information_database:
+            for subcategory in personal_information_database[person.court_role]:
+                for cat_id in personal_information_database[person.court_role][subcategory]:
+                    for db_person in personal_information_database[person.court_role][subcategory][cat_id]:
+                        if set(split_name).issubset(set(db_person['name'].split())):
+                            if not initial or re.search(rf'\s{initial.upper()}\w*', db_person['name']):
+                                person.name = db_person['name']
+                                if db_person.get('gender'):
+                                    person.gender = Gender(db_person['gender'])
+                                if db_person.get('party'):
+                                    person.gender = Gender(db_person['party'])
+                                results.append(person)
+        else:
+            for existing_role in personal_information_database:
+                temp_person = CourtPerson(person.name, court_role=CourtRole(existing_role))
+                db_person, match = match_person_to_database(temp_person, current_gender)
+                if match:
+                    results.append(db_person)
+        if len(results) == 1:
+            if not results[0].gender:
+                results[0].gender = current_gender
+            return person, True
+        return person, False
```

Comment:
If you reused the one from CH_BGer also extract this one in to a function so changes only have to be applied once. 
(Fixes like Line 279, where it should say person.party = PoliticalParty(db_person['party'])

Reply:
I extracted this into a function so that it can be used in several spiders and fixed the line `person.party = PoliticalParty(db_person['party']`.